subprojects/logging/src/main/java/org/gradle/internal/logging/sink/GroupingProgressLogEventGenerator.java

```diff
@@ -84,6 +103,19 @@ private void onStart(ProgressStartEvent startEvent) {
             if (isGroupedOperation(startEvent.getBuildOperationCategory())) {
                 String header = startEvent.getLoggingHeader() != null ? startEvent.getLoggingHeader() : startEvent.getDescription();
                 operationsInProgress.put(buildOpId, new OperationGroup(startEvent.getCategory(), header, startEvent.getTimestamp(), startEvent.getBuildOperationId()));
+
+                if (future == null || future.isCancelled()) {
+                    future = executor.scheduleWithFixedDelay(new Runnable() {
+                        @Override
+                        public void run() {
+                            for (OperationGroup group : operationsInProgress.values()) {
+                                if ((group.lastUpdateTime + LONG_RUNNING_TASK_OUTPUT_FLUSH_TIMEOUT) < timeProvider.getCurrentTime()) {
```

Comment:
I went with time only at this time as it was the easiest one. I would suggest improving this with follow-up PRs. We can do the number of logs and also the size of the buffered logs.